File: Source Code/ManagedDemo/Source/PerfInfo.cpp
#include "stdafx.h"
#include "PerfInfo.h"
#include <fstream>


PerfInfo::PerfInfo(void)
{
	timeSoFar = 0.0;
	times = std::vector<double>(100000);
	for (int i = 0; i < 100000; ++i)
	{
		times[i] = 0.0;
	}
	times[0] = timeSoFar;
	frameCount = 1;
}


PerfInfo::~PerfInfo(void)
{
}

void PerfInfo::Tick( double ms )
{
	timeSoFar += ms;
	if (frameCount < 99999)
		times[frameCount++] = timeSoFar;
	else
		times.push_back(timeSoFar);
}

void PerfInfo::StopAndPrint()
{
	std::ofstream outputFile;
	outputFile.open("./NurseryCollectionTimes.csv");
	for (int i = 0; i < times.size(); ++i)
	{
		outputFile << i << "," << times[i] << std::endl;
	}
	outputFile.close();
}

File: Source Code/.Net Demo/Source/PerfInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace DotNetDemo
{
    public class PerfInfo
    {
        public double timeSoFar;
        List<double> times;

        public PerfInfo()
        {
            timeSoFar = 0.0;
            times = new List<double>(80000);
            times.Add(timeSoFar);
        }

        public void Tick(double ms)
        {
            timeSoFar += ms;
            times.Add(timeSoFar);
        }

        public void StopAndPrint()
        {
            using (System.IO.StreamWriter file = new System.IO.StreamWriter(@"./DotNetDemoTimes.csv"))
            {
                for (int i = 0; i < times.Count; ++i)
                {
                    file.WriteLine(i + "," + times[i]);
                }
            }
        }
    }
}
